include/irr/asset/IAssetManager.h

```diff
@@ -666,7 +666,7 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
                 _outs << "\tKey: " << wtr.first << ", Value: " << static_cast<void*>(wtr.second) << '\n';
         }
 
-        void restoreDummyAsset(SAssetBundle& _bundle)
+        void restoreDummyAsset(SAssetBundle& _bundle, uint32_t _levelsBelow = 0u)
```

Comment:
this function is dangerous, it takes no limits/constraints on asset type to be restored (you will usually know the type of asset you're after).

The amount of duplication should be not top level, but top level + `_levelsBelow`

Also handling of mixed dummy and non dummy assets in a bundle is "meh".

Also no check that length of new bundle == old bundle

Reply:
> this function is dangerous, it takes no limits/constraints on asset type to be restored

asset bundle can hold only one type of assets if thats what you mean

Reply:
no that's not what I meant, I mean that you pass a bundle of some type, then use the bundle's cache key value to try and get a fresh load (badly because only top level will be duplicated and for restore you need the full DAG).

Then you dont limit the asset types you will retrieve/load from `getAssetInHierarchy` so the bundle you might get back could be a different length and even type.

Furthermore you `assert` and not fail gracefully if you dont get same size bundle.

Then you restore without extra checks.

Its basically a surefire way to program termination either via `assert` or trying to restore assets from different types.

Basically the usefulness of this method approaches 0